cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+			}
+		}
+
+		return nil
+	}
+
+	return cfDeploy(config, deployStatement, additionalEnvironment, stopOldAppIfRunning, cmd)
+}
+
+func getManifest(name string) (cloudfoundry.Manifest, error) {
+	return cloudfoundry.ReadManifest(name)
+}
+
+func getAppNameOrFail(config *cloudFoundryDeployOptions) (string, error) {
+
+	if len(config.AppName) > 0 {
+		return config.AppName, nil
+	}
+	if config.DeployType == "blue-green" {
+		return "", fmt.Errorf("Blue-green plugin requires app name to be passed (see https://github.com/bluemixgaragelondon/cf-blue-green-deploy/issues/27)")
+	}
+	if len(config.Manifest) == 0 {
+		return "", fmt.Errorf("Manifest file not provided in configuration. Cannot retrieve app name")
+	}
+	fileExists, err := fileUtils.FileExists(config.Manifest)
+	if err != nil {
+		return "", err
+	}
+	if !fileExists {
+		return "", fmt.Errorf("Manifest file '%s' not found. Cannot retrieve app name", config.Manifest)
+	}
+	m, err := _getManifest(config.Manifest)
+	if err != nil {
+		return "", err
+	}
+	apps, err := m.GetApplications()
+	if err != nil {
+		return "", err
+	}
+	if len(apps) > 0 {
+		namePropertyExists, err := m.ApplicationHasProperty(0, "name")
+		if err != nil {
+			return "", err
+		}
+		if namePropertyExists {
+			appName, err := m.GetApplicationProperty(0, "name")
+			if err != nil {
+				return "", err
+			}
+			if name, ok := appName.(string); ok {
+				if len(name) > 0 {
+					return name, nil
+				}
+			}
+		}
+	}
+
+	return "", fmt.Errorf("No appName available in manifest '%s'", config.Manifest)
+}
+
+func handleSmokeTestScript(smokeTestScript string) ([]string, error) {
+
+	if smokeTestScript == "blueGreenCheckScript.sh" {
+		// what should we do if there is already a script with the given name? Should we really overwrite ...
+		err := fileUtils.FileWrite(smokeTestScript, []byte(defaultSmokeTestScript), 0755)
+		if err != nil {
+			return []string{}, err
+		}
+		log.Entry().Debugf("smoke test script '%s' has been written.", smokeTestScript)
+	}
+
+	if len(smokeTestScript) > 0 {
+		err := fileUtils.Chmod(smokeTestScript, 0755)
+		if err != nil {
+			return []string{}, err
+		}
+		pwd, err := fileUtils.Getwd()
+		if err != nil {
+			return []string{}, err
+		}
+
+		return []string{"--smoke-test", fmt.Sprintf("%s/%s", pwd, smokeTestScript)}, nil
+	}
+	return []string{}, nil
+}
+
+func prepareBlueGreenCfNativeDeploy(config *cloudFoundryDeployOptions) (string, []string, []string, error) {
+
+	smokeTest, err := handleSmokeTestScript(config.SmokeTestScript)
+	if err != nil {
+		return "", []string{}, []string{}, err
+	}
+
+	var deployOptions = []string{}
+
+	if !config.KeepOldInstance {
+		deployOptions = append(deployOptions, "--delete-old-apps")
+	}
+
+	if len(config.Manifest) > 0 {
+		manifestFileExists, err := fileUtils.FileExists(config.Manifest)
+		if err != nil {
+			return "", []string{}, []string{}, err
+		}
+
+		if !manifestFileExists {
+
+			log.Entry().Infof("Manifest file '%s' does not exist", config.Manifest)
```

Comment:
Seems strange: An error during checking if the file exists is bubbled up, but when the file actually does not exist, it is only an informational log entry.

Reply:
No that is not strange. It is a normal situation that a file is not present. But we are in severe trouble if we can't check if a file is present or not. That is worth bubbling up anyway. In such a case it is better not to continue doing anything.

Reply:
ok